### PR TITLE
Fix deprecation warning for Silex 2.3.0

### LINKED ISSUE #518: Incomapatibility with silex v2.3.0
### Description
I recently updated to the latest version of silex as our last update before migrating off of the deprecated silex.

I discovered that silex 2.3.0 updates to symfony/http-foundation v4.

In symfony/http-foundation v4.1.0 they deprecated $request->getSession unless the session already exists, instead they say to check for $request->hasSession.

### Issue
This update of dependencies causes a deprecation warning (which in silex-api those are often converted into fatal exceptions)
### Environment

* bugsnag-php version: 3.16.0
* PHP version: 7.2
* What web framework are you using, if any?: silexphp/Silex/
* How are you installing bugsnag-php? (Composer, phar files?): composer
* Operating system name and version: ubuntu 

### Example

This issue triggers on all silex requests, as long as you don't have error_reporting set to ^E_DEPRECAT

### Solution

I have a patch file with 1 line that would fix this and allow this bugsnag integration to work with silex 2.3.0 for those that are not able to migrate / stuck on silex for whatever reason:

```--- vendor/bugsnag/bugsnag-silex/src/Request/SilexRequest.php       2019-02-04 16:13:24.652968460 -0500
+++ vendor/bugsnag/bugsnag-silex/src/Request/SilexRequest.php   2019-02-04 16:12:53.138949192 -0500
@@ -43,7 +43,7 @@
      */
     public function getSession()
     {
-        $session = $this->request->getSession();
+        $session = $this->request->hasSession() ? $this->request->getSession() : false;

         return $session ? $session->all() : [];
     }
```

### PR DESCRIPTION
**Pull request by tomlongridge**
## Goal

Fixes deprecation warning when calling `request->getSession` if the session doesn't exist. (This is the same fix as was applied in bugsnag/bugsnag-symfony#79.)

### Changed

* SilexRequest - guarded call to `getSession`

## Tests

Re-ran example project with Silex 2.3.

### Linked issues

Fixes bugsnag/bugsnag-php#518

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

**Review by mikeewhite — APPROVED** [review]
LGTM
